tedana/utils.py

```diff
@@ -49,37 +49,70 @@ def reshape_niimg(data):
     return fdata
 
 
-def make_adaptive_mask(data, mask=None, getsum=False, threshold=1):
-    """
-    Make map of `data` specifying longest echo a voxel can be sampled with.
+def make_adaptive_mask(data, mask=None, threshold=1):
+    """Make map of `data` specifying longest echo a voxel can be sampled with.
 
     Parameters
     ----------
     data : (S x E x T) array_like
-        Multi-echo data array, where `S` is samples, `E` is echos, and `T` is
-        time
+        Multi-echo data array, where `S` is samples, `E` is echos, and `T` is time.
     mask : :obj:`str` or img_like, optional
         Binary mask for voxels to consider in TE Dependent ANAlysis. Default is
         to generate mask from data with good signal across echoes
-    getsum : :obj:`bool`, optional
-        Return `masksum` in addition to `mask`. Default: False
     threshold : :obj:`int`, optional
         Minimum echo count to retain in the mask. Default is 1, which is
         equivalent not thresholding.
 
     Returns
     -------
     mask : (S,) :obj:`numpy.ndarray`
-        Boolean array of voxels that have sufficient signal in at least one
-        echo
+        Boolean array of voxels that have sufficient signal in at least ``threshold`` echos.
     masksum : (S,) :obj:`numpy.ndarray`
-        Valued array indicating the number of echos with sufficient signal in a
-        given voxel. Only returned if `getsum = True`
+        Valued array indicating the number of echos with sufficient signal in a given voxel.
+
+    Notes
+    -----
+    The adaptive mask is constructed from two methods:
+
+    1.  Count the total number of echoes in each voxel that have "good" data.
+        This method assumes that an exemplar voxel's signal at later echoes is also reasonable,
+        and that any voxels whose values at a given echo are less than 1/3 of the exemplar voxel's
+        values at that echo are affected by dropout.
+
+        This method uses distributions of values across the mask.
+        Therefore, it is sensitive to the quality of the mask;
+        a bad mask may result in a bad adaptive mask.
```

Comment:
Wait a second... it doesn't look like the adaptive mask is limited to the mask. That seems like a major issue, right?

Reply:
Most of the voxels in the data array are going to be outside the brain.

Reply:
um... You're right. If the function is given a mask, the percentile calculations should only be done within the mask. and the 33rd percentile would be unnecessarily low. I'd call this a bug.

This is removing voxels with a mean of 0, but, if the data are unmasked, it's not removing much. If there's a user-provided mask, it should be applied as part of this line.

https://github.com/ME-ICA/tedana/blob/a21d65ec5b1682a52cb8fe0eee1689d5eb3154a3/tedana/utils.py#L86

The rest of the function should be fine since the adaptive mask criteria can be applied to all voxels and masked later. It's only the percentile threshold that needs to be restricted by the mask.

Reply:
I strongly suspect this is also why we are seeing so many additional voxels being rejected by the new non-monotonic criterion vs the threshold. The percentile threshold is missing voxels outside of the brain and those are the ones flagged by the new threshold.

This means I'm less concerned about comparing old to new masks (and adding temporary options to compare different criteriea), but, given these are still arbitrary criteria, it would still be good to look at the revised adaptive masks on a bunch of data to make sure nothing undesired is happening.

Reply:
I'll open a separate issue about this then.